frontend/src/pages/FundsPullPreApprovals.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useState } from "react";
+import { Approval } from "../interfaces/approval";
+import BackendClient from "../services/backendClient";
+import TestnetWarning from "../components/TestnetWarning";
+import Breadcrumbs from "../components/Breadcrumbs";
+import { Container } from "reactstrap";
+import FundsPullPreApprovalsList from "../components/FundsPullPreApproval/FundsPullPreApprovalsList";
+
+const REFRESH_APPROVALS_INTERVAL = 3000;
+
+function FundsPullPreApprovals() {
+  const [newApprovals, setNewApprovals] = useState<Approval[]>([]);
+  const [activeApprovals, setActiveApprovals] = useState<Approval[]>([]);
+  const [historyApprovals, setHistoryApprovals] = useState<Approval[]>([]);
+
+  let refreshApprovals = true;
```

Comment:
this will be set upon each render, you probably need `useMemo` or `useRef` to prevent it from changing on each render

Reply:
you will have to help me with that